agent/config/runtime_test.go

```diff
@@ -4957,12 +4940,19 @@ func testConfig(t *testing.T, tests []configTest, dataDir string) {
 				if tt.patchActual != nil {
 					tt.patchActual(&actual)
 				}
-				require.Equal(t, expected, actual)
+				assertDeepEqual(t, expected, actual, cmpopts.EquateEmpty())
 			})
 		}
 	}
 }
 
+func assertDeepEqual(t *testing.T, x, y interface{}, opts ...cmp.Option) {
```

Comment:
We discussed putting this is a more centralized place - maybe best to do it in this PR first?